codalab/apps/newsletter/tasks.py

```diff
@@ -0,0 +1,11 @@
+from celery import task
+
+from apps.newsletter.models import NewsletterSubscription
+
+
+@task(queue='site-worker', soft_time_limit=60 * 5)
+def retry_mailing_list():
```

Comment:
Just thought of things we need to do for this. 

We need to check and make sure we have mailchimp settings. Even though this is done in the subscribe/unsubscribe methods, we should do it here before we start hitting the database.

Also, if we do have mailchimp settings, we should fire off a `get` request to mailchimp and make sure we get an `ok` status back so we don't make a bunch of `patch`/`post` requests that are just going to error out.